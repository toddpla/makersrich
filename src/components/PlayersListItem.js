import React, { Component } from 'react';

class PlayersListItem extends Component {

  render() {
    return (
      <div>
        {this.props.displayName}
        <img style={{height: "10px"}}src={`images/${this.props.state}.png`} />
      </div>
    );
  }

}

export default PlayersListItem;
