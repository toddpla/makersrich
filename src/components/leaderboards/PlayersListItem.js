import React, { Component } from 'react';

class PlayersListItem extends Component {

  render() {
    return (
      <tr>

        <td><img style={{height: "10px"}}src={`images/${this.props.state}.png`}/></td>
        <td>{this.props.displayName}</td>
        <td>{this.props.cash}</td>
        <td>{this.props.inventory && this.props.inventory.ruby && this.props.inventory.ruby.length}</td>
        <td>{this.props.inventory && this.props.inventory.bean && this.props.inventory.bean.length}</td>
      </tr>
    );
  }

}

export default PlayersListItem;
