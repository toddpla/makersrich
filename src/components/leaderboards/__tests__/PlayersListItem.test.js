import React from 'react'
import { shallow } from 'enzyme'
import playersData from '../../../test/fixtures/players'
import { PlayersListItem } from '../PlayersListItem'

let player, wrapper;

beforeEach(() => {
  player = playersData.fixtures[0]
  wrapper = shallow(<PlayersListItem key="1" {...player} />)
})

test('should render correctly', () => {
  expect(wrapper).toMatchSnapshot()
})
