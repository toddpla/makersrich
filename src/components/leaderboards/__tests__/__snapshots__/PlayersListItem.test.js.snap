// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayersListItem
    cash={0}
    displayName="Frankie"
    left={0}
    level={1}
    state="online"
    top={0}
    uid="1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td
          className="leaderboard-table-cell"
        >
          <img
            alt="status"
            src="images/online.png"
            style={
              Object {
                "height": "10px",
              }
            }
          />
        </td>,
        <td
          className="leaderboard-table-cell table-displayName"
        >
          Frankie
        </td>,
        <td
          className="leaderboard-table-cell"
        >
          1
        </td>,
        <td
          className="leaderboard-table-cell"
        >
          0
        </td>,
        <td
          className="leaderboard-table-cell"
        />,
        <td
          className="leaderboard-table-cell"
        />,
      ],
      "className": "leaderboard-table-row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="status"
            src="images/online.png"
            style={
              Object {
                "height": "10px",
              }
            }
          />,
          "className": "leaderboard-table-cell",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "status",
            "src": "images/online.png",
            "style": Object {
              "height": "10px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Frankie",
          "className": "leaderboard-table-cell table-displayName",
        },
        "ref": null,
        "rendered": "Frankie",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 1,
          "className": "leaderboard-table-cell",
        },
        "ref": null,
        "rendered": 1,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0,
          "className": "leaderboard-table-cell",
        },
        "ref": null,
        "rendered": 0,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "leaderboard-table-cell",
        },
        "ref": null,
        "rendered": null,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "leaderboard-table-cell",
        },
        "ref": null,
        "rendered": null,
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td
            className="leaderboard-table-cell"
          >
            <img
              alt="status"
              src="images/online.png"
              style={
                Object {
                  "height": "10px",
                }
              }
            />
          </td>,
          <td
            className="leaderboard-table-cell table-displayName"
          >
            Frankie
          </td>,
          <td
            className="leaderboard-table-cell"
          >
            1
          </td>,
          <td
            className="leaderboard-table-cell"
          >
            0
          </td>,
          <td
            className="leaderboard-table-cell"
          />,
          <td
            className="leaderboard-table-cell"
          />,
        ],
        "className": "leaderboard-table-row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="status"
              src="images/online.png"
              style={
                Object {
                  "height": "10px",
                }
              }
            />,
            "className": "leaderboard-table-cell",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "status",
              "src": "images/online.png",
              "style": Object {
                "height": "10px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Frankie",
            "className": "leaderboard-table-cell table-displayName",
          },
          "ref": null,
          "rendered": "Frankie",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 1,
            "className": "leaderboard-table-cell",
          },
          "ref": null,
          "rendered": 1,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
            "className": "leaderboard-table-cell",
          },
          "ref": null,
          "rendered": 0,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "leaderboard-table-cell",
          },
          "ref": null,
          "rendered": null,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "leaderboard-table-cell",
          },
          "ref": null,
          "rendered": null,
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
